express-api/src/services/notifications/notificationServices.ts

```diff
@@ -481,7 +489,48 @@ const updateNotificationStatus = async (notificationId: number, user: User) => {
     query.release();
     return updatedNotification;
   } else if (typeof statusResponse?.status === 'number') {
-    //If we get number type then this wound up being some HTTP code.
+    // handle 404, 422 code here....
+    switch (statusResponse.status) {
+      case 404: {
+        notification.Status = NotificationStatus.NotFound;
+        notification.UpdatedOn = new Date();
+        notification.UpdatedById = user.Id;
+        const updatedNotification = await query.manager.save(NotificationQueue, notification);
+        logger.error(`Notification with id ${notificationId} not found on CHES.`);
+        query.release();
+        return updatedNotification;
+      }
+      case 422: {
+        notification.Status = NotificationStatus.Invalid;
+        notification.UpdatedOn = new Date();
+        notification.UpdatedById = user.Id;
+        const updatedNotification = await query.manager.save(NotificationQueue, notification);
+        logger.error(
+          `Notification with id ${notificationId} could not be processed, some of the data could be formatted incorrectly.`,
+        );
+        query.release();
+        return updatedNotification;
+      }
+      case 401: {
+        notification.Status = NotificationStatus.Unauthorized;
+        notification.UpdatedOn = new Date();
+        notification.UpdatedById = user.Id;
+        const updatedNotification = await query.manager.save(NotificationQueue, notification);
+        logger.error(
+          `Cannot authorize the request to the CHES server, check your CHES credentials.`,
+        );
+        query.release();
+        return updatedNotification;
+      }
```

Comment:
Similar to the comment on the statuses, I don't think we should be changing the status of the notification if we get back a 422 or a 401. Logging it makes sense, but perhaps in that case we just shouldn't update the notification.